private/pkg/app/appcmd/webpages.go

```diff
@@ -0,0 +1,305 @@
+	}
+	defer file.Close()
+	return generateMarkdownPage(cmd, file, slugPrefix)
+}
+
+// generateMarkdownPage creates custom markdown output.
+func generateMarkdownPage(cmd *cobra.Command, w io.Writer, slugprefix string) error {
+	var err error
+	p := func(format string, a ...any) {
+		_, err = w.Write([]byte(fmt.Sprintf(format, a...)))
+	}
+	p("---\n")
+	p("id: %s\n", pageID(cmd))
+	p("title: %s\n", cmd.CommandPath())
+	p("sidebar_label: %s\n", pageName(cmd))
+	p("sidebar_position: %d\n", order(cmd))
+	p("slug: /%s\n", path.Join(slugprefix, slug(cmd)))
+	p("---\n")
+	cmd.InitDefaultHelpCmd()
+	cmd.InitDefaultHelpFlag()
+	if cmd.Version != "" {
+		p("version `%s`\n\n", cmd.Version)
+	}
+	p(cmd.Short)
+	p("\n\n")
+	if cmd.Runnable() {
+		p("### Usage\n")
+		p("```terminal\n$ %s\n```\n\n", cmd.UseLine())
+	}
+	if len(cmd.Long) > 0 {
+		p("### Description\n\n")
+		p("%s \n\n", escapeDescription(cmd.Long))
+	}
+	if len(cmd.Example) > 0 {
+		p("### Examples\n\n")
+		p("```\n%s\n```\n\n", escapeDescription(cmd.Example))
+	}
+	commandFlags := cmd.NonInheritedFlags()
+	commandFlags.SetOutput(w)
+	if commandFlags.HasAvailableFlags() {
+		p("### Flags\n\n")
+		p("```\n")
+		commandFlags.PrintDefaults()
+		p("```\n\n")
+	}
+	inheritedFlags := cmd.InheritedFlags()
+	inheritedFlags.SetOutput(w)
+	if inheritedFlags.HasAvailableFlags() {
+		p("### Flags inherited from parent commands\n\n```\n")
+		inheritedFlags.PrintDefaults()
+		p("```\n\n")
+	}
+	if hasSubCommands(cmd) {
+		p("### Subcommands\n\n")
+		children := cmd.Commands()
+		for _, child := range children {
+			if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
+				continue
+			}
+			p("* [%s](%s/%s)\t - %s\n", child.CommandPath(), cmd.Name(), child.Name(), child.Short)
+		}
+		p("\n")
+	}
+	if cmd.HasParent() {
+		p("### Parent Command\n\n")
+		parent := cmd.Parent()
+		parentName := parent.CommandPath()
+		p("* [%s](../%s)\t - %s\n", parentName, parent.Name(), parent.Short)
+		cmd.VisitParents(func(c *cobra.Command) {
+			if c.DisableAutoGenTag {
+				cmd.DisableAutoGenTag = c.DisableAutoGenTag
+			}
+		})
+	}
+	return err
+}
+
+// commandFilePath converts a cobra command to a path. It stutters the folders and paths
+// in order to allow for rendering of the full command in Docusaurus: "buf/buf beta" for example.
+// Spaces are used in paths because the current version of Docusaurus
+// does not allow for configuring category index pages.
+// This function should be removed when migration off docusaurus occurs.
+func commandFilePath(cmd *cobra.Command) string {
+	commandPath := strings.Split(cmd.CommandPath(), " ")
+	var parentPath, currentPath []string
+	for i := range commandPath {
+		currentPath = append(parentPath, strings.Join(commandPath[:i+1], " "))
+		parentPath = currentPath
+	}
+	fullPath := path.Join(currentPath...)
+	if cmd.HasSubCommands() {
+		return path.Join(fullPath, "index.md")
+	}
+	return fullPath + ".md"
+}
+
+func order(cmd *cobra.Command) int {
```

Comment:
orderWhat? all these functions are now global to appcmd, so make their naming more specific

Reply:
Resolved: 40b789bef98b7419f02edadc56b84e61157e685b